<launch>
	<arg name="node_name" value="turbo_node"/>
	<arg name="veh" doc="Name of vehicle. ex: megaman"/>
    <arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman"/>

	<group ns="$(arg veh)">
		<!-- Remap wheels command topic -->
		<remap from="$(arg node_name)/wheels_cmd_executed" to="wheels_driver_node/wheels_cmd_executed"/>
		<!-- Remap wheel encoder topics -->
		<remap from="$(arg node_name)/left_wheel_encoder/tick" to="left_wheel_encoder_node/tick"/>
		<remap from="$(arg node_name)/right_wheel_encoder/tick" to="right_wheel_encoder_node/tick"/>
		<!-- Remap gain parameter -->
		<remap from="$(arg node_name)/gain" to="kinematics_node/gain"/>
		<!-- Node -->
	    <node pkg="turbo" type="turbo_node.py"  name="$(arg node_name)" output="screen" required="true">
			<rosparam command="load" file="$(find turbo)/config/$(arg node_name)/$(arg param_file_name).yaml" />
		</node>
	</group>

</launch>
